include/sys/multi_heap.h

```diff
@@ -125,6 +132,17 @@ void *sys_multi_heap_alloc(struct sys_multi_heap *mheap, void *cfg, size_t bytes
 void *sys_multi_heap_aligned_alloc(struct sys_multi_heap *mheap,
 				   void *cfg, size_t align, size_t bytes);
 
+/**
+ * @brief Get a specific heap for provided address
+ *
+ * Returns the pointer to a single system heap (with metadata)
+ *  controlling the provided pointer
+ *
+ * @param mheap Multi heap pointer
+ * @param addr address to be found
+ */
+struct sys_heap_ext *sys_multi_heap_get_heap(struct sys_multi_heap *mheap, void *addr);
```

Comment:
Returning pointers into internally-managed data structures is sort of a dangerous pattern.  Suggest making this const.

Alternatively, since it's just a two-word struct, you could pass it as an argument and have the multiheap code fill it in for the caller.  That would open the possibility of returning an error code like -ENOENT on bad data.